### PR TITLE
Add support for OSD_EFFICIENCY

### PR DESCRIPTION
**Pull request by fiam**
Shows current efficiency as mah/km. Show also average efficiency
in the stats screen after disarming.

PR for configurator https://github.com/iNavFlight/inav-configurator/pull/274

### CODE REVIEW

File: src/main/io/osd.c [hunk 1]
```diff
@@ -1405,6 +1405,27 @@ static bool osdDrawSingleElement(uint8_t item)
...
 
+    case OSD_EFFICIENCY:
+        {
+            // amperage is in centi amps, speed is in cms/s. We want
+            // mah/km. Values over 999 are considered useless and
+            // displayed as "---""
+            const int32_t maxEfficiencyValue = 999;
+            int32_t value = maxEfficiencyValue + 1;
+            if (gpsSol.groundSpeed > 0) {
+                value = (amperage / gpsSol.groundSpeed) / 0.0036f;
```
Comment (digitalentity) [inline src/main/io/osd.c]: You need explicit conversion to `float` here. `amperage` and `gpsSol.groundSpeed` are integer. For efficiency below 278 mah/km `amperage / gpsSol.groundSpeed` will yield zero.

**Review by digitalentity — CHANGES_REQUESTED** [review]

File: src/main/io/osd.c (hunk 1 above)
Comment (fiam) [reply src/main/io/osd.c]: I hadn't realised... what if we multiply `amperage` by `1/0.0036` (277.77, so we either pick 277 or 278) so we can remove the floating point math?

Comment (digitalentity) [reply src/main/io/osd.c]: Yeah, you can change to `value = 278 * amperage / gpsSol.groundSpeed;` Less than 0.1% error is marginal.

Comment (fiam) [reply src/main/io/osd.c]: Ok, let's do that

**Review by digitalentity — APPROVED** [review]

File: src/main/io/osd.c [hunk 2]
```diff
@@ -1628,6 +1670,16 @@ static void osdShowStats(void)
         itoa(mAhDrawn, buff, 10);
         strcat(buff, "\x07");
         displayWrite(osdDisplayPort, statValuesX, top++, buff);
+
+        if (STATE(GPS_FIX)) {
```
Comment (digitalentity) [inline src/main/io/osd.c]: Should we check here? Isn't non-zero `totalDistance` a sign that we did have GPS at least some of the trip?

**Comment by fiam** [comment]
@digitalentity Please, kick travis to restart on this one. Failed due to not having enough disk space.

**Comment by giacomo892** [comment]
@fiam just push it again (just add a space in a comment) and it will run again :D

**Comment by fiam** [comment]
@giacomo892 Thanks for the idea! Just realised that if I altered the branch, travis would probably run again. So I rebased before the last commit, force pushed, cherry-picked the last commit again and force pushed once more. It seems that kicked the tests again 😄

**Comment by fiam** [comment]
Unfortunately, it seems the travis failed again due to not having space on the VM disk. Maybe we should remove the temporary files for each target after building it?

**Pull request merged**